### PR TITLE
[INLONG-9774][Sort] Support rowdata way of sort InLong message tlog-kv format

### LINKED ISSUE #9774: [Feature][Sort] Support rowdata way of sort InLong message tlog-kv format
### Description

such as in log file:
streamId|k1=v1&k2=v2

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
- Fixes #9774

### CODE REVIEW

File: inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/test/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public void testInvalidIgnoreParseErrors() {
+        addPropertyAndVerify(CUSTOM_DESCRIPTOR_WITH_SCHEMA, "format.escape-character", "DDD");
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testMissingSchema() {
+        removePropertyAndVerify(CUSTOM_DESCRIPTOR_WITH_SCHEMA, "format.schema");
+    }
+
+    // --------------------------------------------------------------------------------------------
```
Comment (aloyszhang) [inline inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/test/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvTest.java:81]: remove this line

**Comment by aloyszhang** [comment]
Please fix the failed tests https://github.com/apache/inlong/actions/runs/8166147677/job/22324424993?pr=9780

File: inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/pom.xml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        <version>1.12.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>format-inlongmsg-rowdata-tlogkv</artifactId>
+
+    <packaging>jar</packaging>
+    <name>Apache InLong - Sort Format-InLongMsg-Rowdata-TLogKV</name>
+
+    <properties>
+        <inlong.root.dir>${project.parent.parent.parent.parent.basedir}</inlong.root.dir>
```
Comment (aloyszhang) [inline inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/pom.xml:37]: Sorry, but I did not find the place this property used.

File: inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/test/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvTest.java (hunk 1 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/test/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvTest.java]: done

File: inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/pom.xml (hunk 2 above)
Comment (baomingyu) [reply inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/pom.xml:37]: code check style use

**Review by aloyszhang — DISMISSED** [review]

File: inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/main/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvFormatDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+                        delimiter,
+                        entryDelimiter,
+                        kvDelimiter,
+                        escapeChar,
+                        quoteChar));
+    }
+
+    @Override
+    protected List<RowData> convertRowDataList(InLongMsgHead head, InLongMsgBody body) throws Exception {
+        GenericRowData rowD =
```
Comment (EMsnap) [inline inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/main/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvFormatDeserializer.java:209]: rowD is vague, maybe genericData ? or genericRow

Comment (baomingyu) [reply inlong-sort/sort-formats/format-rowdata/format-inlongmsg-rowdata-tlogkv/src/main/java/org/apache/inlong/sort/formats/inlongmsgtlogkv/InLongMsgTlogKvFormatDeserializer.java:209]: done change to genericRowData

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**